### PR TITLE
Added support for compressed tfrecord files

### PR DESCRIPTION
**Pull request by nanton96**

### CODE REVIEW

File: deephub/models/feeders/tfrecords/__init__.py [hunk 1]
```diff
@@ -246,6 +248,14 @@ def get_tf_dataset(self, epochs: Optional[int] = None) -> tf.data.Dataset:
         # buffer_size=None is autotuning the appropriate prefetching buffer size
         return batches.prefetch(buffer_size=self.prefetch)
```
Comment (Katsileros) [inline deephub/models/feeders/tfrecords/__init__.py]: The extension for the files should be as an input from the config as we discussed. I don't like the fact that we have to scan the 1st file in order to find the extension type. Also, (I am not 100% sure), but if not all files are with the same extension maybe by default the tensor flow API throws an extension. You can check it.

File: deephub/models/feeders/tfrecords/meta.py [hunk 2]
```diff
@@ -54,9 +54,11 @@ def generate_for(cls, fpath: Path) -> 'TFRecordFileInfo':
 
         """
         file_size = fpath.stat().st_size
```
Comment (Katsileros) [inline deephub/models/feeders/tfrecords/meta.py]: The compression type here, must also be an argument to this function. It must be passed as a command line argument from the cli. Add a new argument, compression_type, with default value equal to empty.

Comment (Katsileros) [inline deephub/models/feeders/tfrecords/meta.py]: The compression type here, must also be an argument to this function. It must be passed as a command line argument from the cli. Add a new argument, compression_type, with default value equal to empty.

**Comment by nanton96** [comment]
In case you give some unzipped files generate_metadata will skip them with the following error:
```
inflate() failed with error -3: incorrect header check
```

**Comment by Katsileros** [comment]
> In case you give some unzipped files generate_metadata will skip them with the following error:
> 
> ```
> inflate() failed with error -3: incorrect header check
> ```

And it will produce the info for the rest that are zipped or it will fail completely?

**Comment by nanton96** [comment]
It does produce the info for the rest.

**Pull request merged**